lib/internal/modules/esm/hooks.js

```diff
@@ -403,11 +403,7 @@ class Hooks {
 
     let responseURLObj;
     if (typeof responseURL === 'string') {
-      try {
-        responseURLObj = new URL(responseURL);
-      } catch {
-        // responseURLObj not defined will throw in next branch.
-      }
+      responseURLObj = URL.parse(responseURL);
```

Comment:
This is mutable. I am pretty sure at least in the ESM loader we try not to look things up from mutable public APIs…

Reply:
if I export `URLParse: URL.parse` in `internal/url.js` would it make it immutable?